### PR TITLE
Added sample JupyterHub config

### PR DESCRIPTION
**Pull request by t20100**
This PR adds a `demo_jupyterhub_conf.py` JHub config that serves both as example and can be used to run the server locally for development.

### CODE REVIEW

File: demo_jupyterhub_conf.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import sys
+import batchspawner  # noqa
+
+sys.path.insert(0, os.path.abspath(os.path.dirname(__file__)))
+
+import jupyterhub_moss  # noqa
+
+c = get_config()  # noqa
+
+# Select the MOSlurmSpawner backend and increase the timeout since batch jobs may take time to start
```
Comment (loichuder) [inline demo_jupyterhub_conf.py:14]: The comment about timeout should be removed IMO

File: demo_jupyterhub_conf.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            k: {"nodes": v["max_nprocs"], "idle": random.randint(0, v["max_nprocs"])}
+            for k, v in self.partitions.items()
+        }
+
+
+c.JupyterHub.spawner_class = MOckSlurmSpawner
+
+# Partition descriptions
+c.MOSlurmSpawner.partitions = {
+    "partition_1": {  # Partition name     # (See description of fields below for more info)
```
Comment (loichuder) [inline demo_jupyterhub_conf.py:30]: I would remove these comments with a link to https://github.com/silx-kit/jupyterhub_moss#partition-settings instead

File: demo_jupyterhub_conf.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        "max_ngpus": 0,
+        "max_nprocs": 28,
+        "max_runtime": 12 * 3600,
+        "simple": False,
+        "venv": "/path/to/jupyter/env/for/partition_3/bin/",
+    },
+}
+
+
+# c.Authenticator.auto_login = True
```
Comment (loichuder) [inline demo_jupyterhub_conf.py:64]: To be removed ?

File: demo_jupyterhub_conf.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+# JupyterHub
+c.JupyterHub.ip = "127.0.0.1"
+c.JupyterHub.hub_ip = "127.0.0.1"
+c.JupyterHub.port = 8000
+
+# Batchspawner
+c.BatchSpawnerBase.exec_prefix = ""  # Do not run sudo
+
+# c.ConfigurableHTTPProxy.command = ["configurable-http-proxy", "--redirect-port", "8000"]
```
Comment (loichuder) [inline demo_jupyterhub_conf.py:74]: To be removed ?

**Review by loichuder — APPROVED** [review]

File: demo_jupyterhub_conf.py (hunk 3 above)
Comment (t20100) [reply demo_jupyterhub_conf.py:64]: If not enabled, yes

**Pull request merged**